src/models/predicates.js

```diff
@@ -36,22 +36,24 @@ function normalize (obj, config, encoding) {
         transform = combinators.compose(exceptTransform, caseTransform, encodeTransform),
         transformAll = function (o) {
             if (!o) {
-              return o;
+                return o;
             }
-            return Object.keys(o).reduce(function (result, key) {
-                var value = o[key];
-                if (Array.isArray(o[key])) {
-                    value = o[key].map(transform);
```

Comment:
If the array contained objects(instead of strings) an exception would be thrown when the "toLower()" method was called on the object.